### PR TITLE
Added reducers and actions

### PR DESCRIPTION
**Pull request by Talha-Hanif5666**
### Here I have done these steps in BookStore Web App 👍
- [ ] Add structure using the feature folder approach.
- [ ] Used ducks pattern.
- [ ] Add actions and reducers.
- [ ] Add configuration file for the store.
- [ ] Add Provider in Index.js file.

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      return {
+        ...state,
+        books: updatedBooks,
+      };
+    default:
+      return state;
+  }
+};
+
+export default bookReducer;
```
Comment (thecodechaser) [inline src/redux/books/bookSlice.js]: - [ ] Please make sure to create and export action types for adding and removing a book as requested in the project requirements.

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      return {
+        ...state,
+        status: 'Under construction',
+      };
+    default:
+      return state;
+  }
+};
+
+export default categoryReducer;
```
Comment (thecodechaser) [inline src/redux/categories/categoriesSlice.js]: - [ ] Please make sure to create and export action types for checking the status as requested in the project requirements.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Talha-Hanif5666👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Highly organized code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
**Hi @Talha-Hanif5666
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ Good PR description

## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**